examples/message_passing/after.cpp

```diff
@@ -0,0 +1,57 @@
+// This example shows how to use caf::after
+
+#include "caf/all.hpp"
+
+#include <chrono>
+#include <iostream>
+#include <random>
+
+using std::cout;
+using std::endl;
+
+using namespace caf;
+
+// Sends a random number of printable characters to buddy and exits
+void generator(event_based_actor* self, actor buddy) {
+  std::random_device rd;
+  std::minstd_rand gen{rd()};
+  const auto count = std::uniform_int_distribution<>{20, 100}(gen);
+  std::uniform_int_distribution<> dis{33, 126};
+  for (auto i = 0; i < count; i++) {
+    self->send(buddy, static_cast<char>(dis(gen)));
+  }
+}
+
+// Collects the incoming characters until either the awaited_size of characters
+// is received, or no new characters arrive for 100ms
+behavior collector(stateful_actor<std::string>* self, size_t awaited_size) {
+  using namespace std::chrono_literals;
+  self->state.reserve(awaited_size);
+  return {
+    [=](char c) {
```

Comment:
We don't do this consistently yet, but there are reasons to avoid `[=]`:
- It's greedy.
- C++20 deprecated the implicit capture of `this` via `[=]`. So there's already one case where an explicit capture is required by the language. Just listing all is more consistent.

```suggestion
    [self](char c) {
```

Reply:
I needed `self` and the `awaited_size`, which means everything. Since I've decided to drop the latter self sounds great. Also didn't know about deprecating the capture of `this`.